src/lib/utils/DY.Lib.SplitFunction.fst

```diff
@@ -398,3 +398,28 @@ let rec memP_mk_dependent_tagged_local_funs #tag_set_t #local_fun_t l tag_set lo
   | [] -> ()
   | (x,y)::t ->
     memP_mk_dependent_tagged_local_funs t tag_set local_fun
+
+/// Normalize pre-condition and post-conditions from correctness lemmas derived from `mk_global_fun_correct`
+/// to save some boilerplate when using this framework
+
+val get_fv_name: FStar.Tactics.term -> FStar.Tactics.Tac unit
+let get_fv_name t =
+  let open FStar.Tactics in
+  match t with
+  | Tv_FVar fv -> (
+    exact (Tv_Const (C_String (implode_qn (inspect_fv fv))))
+  )
+  | _ -> fail ("get_fv_name: not a free variable: " ^ (term_to_string t))
+
+val do_boilerplate:
+  #a:Type ->
+  #p:(a -> Type) -> #q:(a -> Type) ->
+  ($lem: (x: a -> Lemma (requires (p x)) (ensures (q x)))) ->
```

Comment:
Minor point: I don't think `$` on arguments is very well-documented (but it may just be that it's hard to search for in the documentation). If this has any relevance to the user of this function, it might be nice to note in a comment what purpose it serves.

Reply:
Done, indeed it is badly documented